pages/api/augment-request-gql.ts

```diff
@@ -0,0 +1,408 @@
+    const requestHash = optimisticPriceRequest.requestHash;
+    const requestLogIndex = optimisticPriceRequest.requestLogIndex;
+    const uniqueKey = optimisticPriceRequest.id;
+    return {
+      time,
+      uniqueKey,
+      identifier,
+      ooRequestUrl: constructOoUiLink(
+        requestHash,
+        chainId,
+        "OptimisticOracleV2",
+        requestLogIndex
+      ),
+      originatingChainTxHash: requestHash,
+      originatingChainId: chainId,
+      originatingOracleType: "OptimisticOracleV2",
+    };
+  } catch (error) {
+    if (debug) console.error("oov2 error:", error);
+    throw error;
+  }
+}
+async function oov1Query({
+  time,
+  identifier,
+  chainId,
+}: OracleQueryParams): Promise<ResponseBody> {
+  const subgraph = getSubgraphConfig("Optimistic Oracle V1", chainId);
+  const query = gql`
+    query oov1Query($proposalTimestamp: Int!) {
+      optimisticPriceRequests(
+        where: { proposalTimestamp: $proposalTimestamp }
+      ) {
+        id
+        requestHash
+        requestLogIndex
+        requestTimestamp
+      }
+    }
+  `;
+  type GqlRequest = {
+    id: string;
+    requestHash: string;
+    requestLogIndex: string;
+  };
+  type GqlResponse = {
+    optimisticPriceRequests: GqlRequest[];
+  };
+
+  try {
+    const data: GqlResponse = await request(subgraph.url, query, {
+      proposalTimestamp: time,
+    });
+    assert(data.optimisticPriceRequests.length > 0, "request not found");
+    const [optimisticPriceRequest] = data.optimisticPriceRequests;
+    const requestHash = optimisticPriceRequest?.requestHash;
+    const requestLogIndex = optimisticPriceRequest?.requestLogIndex;
+    const uniqueKey = optimisticPriceRequest?.id;
+    return {
+      time,
+      uniqueKey,
+      identifier,
+      ooRequestUrl: constructOoUiLink(
+        requestHash,
+        chainId,
+        "OptimisticOracle",
+        requestLogIndex
+      ),
+      originatingChainTxHash: requestHash,
+      originatingChainId: chainId,
+      originatingOracleType: "OptimisticOracle",
+    };
+  } catch (error) {
+    if (debug) console.error("oov1 error:", error);
+    throw error;
+  }
+}
+async function tryAllTypes(request: RequestBody): Promise<ResponseBody> {
+  const chainId =
+    extractChildChainId(decodeHexString(request.ancillaryData)) ?? 1;
+  const results = await Promise.allSettled([
+    oov1Query({ ...request, chainId }),
+    oov2Query({ ...request, chainId }),
+    oov3Query({ ...request, chainId }),
+    ooSkinnyQuery({ ...request, chainId }),
+  ]);
+  const successResult: undefined | PromiseFulfilledResult<ResponseBody> =
+    results.find(
+      (result): result is PromiseFulfilledResult<ResponseBody> =>
+        result.status === "fulfilled"
+    );
+
+  if (!successResult) {
+    throw new Error("All requests failed");
+  }
+
+  return successResult?.value;
+}
+
+async function augmentRequest(request: RequestBody): Promise<ResponseBody> {
```

Comment:
entry point function is a  lot cleaner now. this works by getting the l1 vote data, and figuring out where the chain id of the originating request came from using the ancillary data ( includes child chain id).  Then we query every known type of OO on that chain and hopefully find one that returns valid data.  This lets us know what type of oracle it was requested from.